rllib/agents/sac/apex.py

```diff
@@ -1,35 +1,35 @@
+"""ApexSAC Trainer and default config."""
+
 from ray.rllib.agents.dqn.apex import APEX_TRAINER_PROPERTIES
-from ray.rllib.agents.sac.sac import DEFAULT_CONFIG as SAC_CONFIG, SACTrainer
-from ray.rllib.utils import merge_dicts
+from ray.rllib.agents.sac.sac import SACTrainer, DEFAULT_CONFIG as SAC_CONFIG
 
 # yapf: disable
 # __sphinx_doc_begin__
 
-APEX_SAC_DEFAULT_CONFIG = merge_dicts(
-    SAC_CONFIG,  # see also the options in dqn.py, which are also supported
+APEX_SAC_DEFAULT_CONFIG = SACTrainer.merge_trainer_configs(
+    SAC_CONFIG,  # see also the options in sac.py, which are also supported
     {
-        "optimizer": merge_dicts(
-            SAC_CONFIG["optimizer"], {
-                "max_weight_sync_delay": 400,
-                "num_replay_buffer_shards": 4,
-                "debug": False
-            }),
+        "optimizer": {
+            "max_weight_sync_delay": 400,
+            "num_replay_buffer_shards": 4,
+            "debug": False,
+        },
         "n_step": 1,
         "num_gpus": 0,
         "num_workers": 32,
         "buffer_size": 2000000,
         "learning_starts": 50000,
         "train_batch_size": 512,
         "rollout_fragment_length": 50,
-        "target_network_update_freq": 500000,
+        "target_network_update_freq": 0,
```

Comment:
This implies updating the target model at every step, isn't it?
Just curious, why in SAC we want to do that?

Reply:
Yes. This is the default setting for SAC. For Apex the default setting is 500,000. SAC just uses the target network a bit differently than the other algorithms. In other algorithms, the target network is meant to stabilize the Q Net updates. SAC instead uses twin q networks to do that. We might want to set this value higher if we aren't using twin q networks in SAC.

Reply:
Got it